domain/status/service/service.go

```diff
@@ -748,5 +754,53 @@ func (s *Service) GetModelStatusInfo(ctx context.Context) (status.ModelStatusInf
 	ctx, span := trace.Start(ctx, trace.NameFromFunc())
 	defer span.End()
 
-	return s.st.GetModelStatusInfo(ctx)
+	return s.modelState.GetModelStatusInfo(ctx)
+}
+
+// GetStatus returns the current status of the model.
+//
+// The following error types can be expected to be returned:
+// - [modelerrors.NotFound]: When the model does not exist.
```

Comment:
Suggest a wording change here. By the fact that caller has this service means the model did exist. In my head this sounds like the more reasonable wording.

`- [modelerrors.NotFound] when the model no longer exists.`